otrs/operations.py

```diff
@@ -68,58 +69,84 @@ def update_ticket(config, params):
             update_params.update({"Priority": params["priority"]})
         if params.get("customer"):
             update_params.update({"CustomerUser": params["customer"]})
-        if params.get("dynamicField"):
-            name = params.get("dynamicField")['Name']
-            value = params.get("dynamicField")['Value']
-            df = pyotrs.DynamicField(name, value)
-            update_params.update(dynamic_fields=[df])
-        if params.get("article_subject") and params.get("article_body"):
-            new_article = pyotrs.Article({"Subject": params.get("article_subject"), "Body": params.get("article_body")})
-            update_params.update({"article": new_article})
-
-        ticket_info = client.ticket_update(ticket_id, **update_params)
+        
+        if params["dynamicField"]:
+            dynamicFieldsParameter = params["dynamicField"]
+            dFields = []
+            for item in dynamicFieldsParameter:
+              name = item['Name']
+              value = item['Value']
+              dFields.append(pyotrs.DynamicField(str(name), str(value)))
+            update_params.update(dynamic_fields=dFields)
+            
+
+        ticket_info = client.ticket_update(ticket_id, **update_params )
 
         return {"ticket_metadata": ticket_info}
-    except Exception as Err:
-        raise ConnectorError(Err)
+    except Exception as e:
+        error_message = "Error updating ticket {0}. Error message as follows:\n{1}".format(ticket_id, str(e))
+        logger.exception(error_message)
+        raise ConnectorError(error_message)
 
 
 def get_ticket(config, params):
     try:
         client = _create_client(config)
+    except ConnectorError as e:
```

Comment:
what is purpose of adding try exception block here. create_client is already taking care of it.